src/main/java/codeu/model/store/basic/NotificationStore.java

```diff
@@ -0,0 +1,78 @@
+package codeu.model.store.basic;
+
+import codeu.model.data.Notification;
+import codeu.model.data.Message;
+import codeu.model.data.User;
+import codeu.model.store.persistence.PersistentStorageAgent;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * Store class that uses in-memory data structures to hold values and automatically loads from and
+ * saves to PersistentStorageAgent. It's a singleton so all servlet classes can access the same
+ * instance.
+ */
+ public class NotificationStore {
+
+   /** Singleton instance of NotificationStore. */
+   private static NotificationStore instance;
+
+   /**
+   * Return the singleton instance of NotificationStore that should be shared between all Servlet
+   * classes. Do not call this function from a test; use getTestInstance() instead.
```

Comment:
You need to add a `getTestInstance` function :-). Also, add tests. Somethings you could test:
1. `getNotificationsForUser`
2. `getNumNotificationsForUser`
Both of the above could be tested after calls to `addNotification`. Maybe also add a test to `addNotification`, checking it it actually mutates a dummy datastore. (See if we have other test classes where we deal with dummy DataStore objects).